sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md

```diff
@@ -82,6 +82,98 @@ With this configuration, the Azure Monitor Distro will use the credentials of th
+
+#### Customizing ASP.NET Core Instrumentation Options
+
+```C#
+builder.Services.AddOpenTelemetry().UseAzureMonitor();
+builder.Services.Configure<AspNetCoreInstrumentationOptions>(options =>
+{
+    options.RecordException = true;
+    options.Filter = (httpContext) =>
+    {
+        // only collect telemetry about HTTP GET requests
+        return httpContext.Request.Method.Equals("GET");
+    };
+});
+```
+
+#### Customizing HttpClient Instrumentation Options
+
+```C#
+builder.Services.AddOpenTelemetry().UseAzureMonitor();
+builder.Services.Configure<HttpClientInstrumentationOptions>(options =>
+{
+    options.RecordException = true;
+    options.FilterHttpRequestMessage = (httpRequestMessage) =>
+    {
+        // only collect telemetry about HTTP GET requests
+        return httpRequestMessage.Method.Equals("GET");
+    };
+});
+```
+
+#### Customizing SQLClient Instrumentation Options
+
+```C#
+builder.Services.AddOpenTelemetry().UseAzureMonitor();
+builder.Services.Configure<SqlClientInstrumentationOptions>(options =>
+{
+    options.SetDbStatementForStoredProcedure = false;
+});
+```
+
+#### Customizing Sampling Percentage
+
+```C#
+builder.Services.AddOpenTelemetry().UseAzureMonitor()
+                                   .WithTracing(b =>
+                                   {
+                                       b.SetSampler(new ApplicationInsightsSampler(0.9F));
+                                   });
+```
+
+(or)
+
+```
+builder.Services.ConfigureOpenTelemetryTracerProvider((sp, builder) => builder.SetSampler(new ApplicationInsightsSampler(0.9F)));
+```
+
+#### Adding Custom ActivitySources to Traces
+
+```C#
+builder.Services.AddOpenTelemetry().UseAzureMonitor()
+                                   .WithTracing(b =>
+                                   {
+                                       b.AddSource("MyCompany.MyProduct.MyLibrary");
+                                   });
+```
+
+#### Adding Custom Meter to Metrics
+
+```C#
+builder.Services.AddOpenTelemetry().UseAzureMonitor()
+                                   .WithMetrics(b =>
+                                   {
+                                       b.AddMeter("MyCompany.MyProduct.MyLibrary");
+                                   });
+```
+
+#### Adding Additional Instrumentation
+
+If you need to instrument a library or framework that isn't included in the Azure Monitor Distro, you can add additional instrumentation using the OpenTelemetry Instrumentation packages. For example, to add instrumentation for gRPC clients, you can add the [OpenTelemetry.Instrumentation.GrpcNetClient](https://www.nuget.org/packages/OpenTelemetry.Instrumentation.GrpcNetClient/) package and use the following code:
+
+```C#
+builder.Services.AddOpenTelemetry().UseAzureMonitor()
```

Comment:
it'd be good to mention that irrespective of the number of times this is called, there is single tracer/meter provider only, and every config is simply being applied to that single one. And call out that if additional provider is needed, hand-construction is needed.